discord/automod.py

```diff
@@ -0,0 +1,413 @@
+        return self._state.get_user(self.creator_id)
+
+    @property
+    def exempt_roles(self) -> List[Role]:
+        """List[:class:`Role`]: The roles that are exempt from this rule."""
+        it = [role for role in (self.guild.get_role(role_id) for role_id in self.raw_exempt_roles) if role is not None]
+        return utils._unique(it)
+
+    @property
+    def exempt_channels(self) -> List[Union[GuildChannel, Thread]]:
+        """List[Union[:class:`abc.GuildChannel`, :class:`Thread`]]: The channels that are exempt from this rule."""
+        it = filter(None, map(self.guild._resolve_channel, self.raw_exempt_channels))
+        return utils._unique(it)
+
+    @property
+    def actions(self) -> List[AutoModRuleAction]:
+        """List[:class:`AutoModRuleAction`]: The actions that are taken when this rule is triggered."""
+        return [AutoModRuleAction.from_data(action) for action in self._actions]
+
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = None,
+        event_type: Optional[AutoModRuleEventType] = None,
+        actions: List[AutoModRuleAction],
+        enabled: Optional[bool] = None,
+        exempt_roles: Optional[List[Role]] = None,
+        exempt_channels: Optional[List[Union[GuildChannel, Thread]]] = None,
+    ) -> Self:
+        """|coro|
+
+        This method will alter an existing auto moderation rule.
+
+        Parameters
+        -----------
+        name: Optional[:class:`str`]
+            The name to change to, if any.
+        event_type: Optional[:class:`AutoModRuleEventType`]
+            The event type to change to, if any.
+        actions: List[:class:`AutoModRuleAction`]
+            The rule actions to update, if any.
+        enabled: Optional[:class:`bool`]
+            Whether the rule should be enabled or not.
+        exempt_roles: Optional[List[:class:`Role`]]
+            The roles to exempt from the rule, if any.
+        exempt_channels: Optional[List[Union[:class:`abc.GuildChannel`], :class:`Thread`]]
+            The channels to exempt from the rule, if any.
+
+        Raises
+        -------
+        :exc:`Forbidden`
+            You are not authorised to perform this action.
+        :exc:`NotFound`
+            The rule was not found.
+
+        Returns
+        --------
+        :class:`AutoModRule`
+            The updated auto moderation rule.
+        """
+        transformed_actions = None
+        if actions:
+            transformed_actions = [action.to_dict() for action in actions]
+
+        data = await self._state.http.modify_auto_moderation_rule(
+            self.guild.id,
+            self.id,
+            name=name,
+            event_type=event_type,
+            actions=transformed_actions,
+            enabled=enabled,
+            exempt_roles=exempt_roles,
+            exempt_channels=exempt_channels,
+        )
+
+        return self.__class__(data=data, guild=self.guild, state=self._state)
```

Comment:
Explicit constructor?

Reply:
How exactly would you handle this one?

Reply:
I don't see anything inheriting this (unless we're allowing users to inherit it) so can't we just `AutoModRule(...)`?

Reply:
Huh, I thought that would NameError. Good to know, thanks.